src/Mailgun/Connection/RestClient.php

```diff
@@ -310,6 +310,10 @@ protected function prepareFile($fieldName, $filePath)
             $resource = fopen($filePath, 'r');
         }
 
+        if(is_resource($resource)) {
+            fclose($resource);
```

Comment:
Are you sure you can close it here? The resource is being used after this point.

Reply:
@Nyholm you are right, well the only point then would before to return in [line 179](https://github.com/shakaran/mailgun-php/blob/cb273daa088f03c79d7f35fb90334d9c1b221694/src/Mailgun/Connection/RestClient.php#L179) the problem is that it should force to iterate over all the array $postFiles and pick the field "contents" to close it. So in terms of perfomance, I am not sure if it will degrade it, what do you think?

Reply:
@Nyholm any update? several months has passed and I am still waiting for some feedback/input. it seems that the project has little activity or none commiting new code :(